src/wiki/plugins/metadata/tables.py

```diff
@@ -129,29 +129,28 @@ def tokens_for_supersense(article_id):
     t = PTokenAnnotation.objects.select_related('construal__article__current_revision',
                                                 'construal__role__article__current_revision', 'construal__function__article__current_revision',
                                                 'usage__current_revision__metadatarevision__usagerevision__article_revision__article',
-                                                'sentence', 'adposition').filter(Q(construal__role__article__id=article_id) | Q(construal__function__article__id=article_id)).order_by('sentence__sent_id')
+                                                'sentence', 'adposition').filter(Q(sentence__corpus__deprecated=False), Q(construal__role__article__id=article_id) | Q(construal__function__article__id=article_id)).order_by('sentence__sent_id')
```

Comment:
Is the `Q()` wrapper necessary for the first item? My guess is it's only necessary for the disjunction.

Reply:
Yes, it is necessary. Notice that usually `filter` takes named arguments, but `Q` objects are taken as positional arguments which in python must come first. See [complex-lookups-with-q-objects](https://docs.djangoproject.com/en/3.0/topics/db/queries/#complex-lookups-with-q-objects). You might be able to mix argument types, but I think this is cleaner.